docarray/typing/tensor/abstract_tensor.py

```diff
@@ -61,7 +61,7 @@ class AbstractTensor(Generic[ShapeT], AbstractType, ABC):
 
     @classmethod
     @abc.abstractmethod
-    def __docarray_validate_shape__(cls, t: T, shape: Tuple[int]) -> T:
+    def _docarray_validate_shape(cls, t: T, shape: Tuple[int]) -> T:
```

Comment:
I belive this should keep the old name, because it legitimately is a magic method that gets called under the hood, very similar to python built-in magic methods